src/bandwidth.rs

```diff
@@ -134,21 +130,29 @@ impl<TInner: TryFuture> Future for BandwidthFuture<TInner> {
     }
 }
 
-/// Allows obtaining the average bandwidth of the connections created from a `BandwidthLogging`.
+/// Allows obtaining the average bandwidth of the connections created from a [`BandwidthLogging`].
 pub struct BandwidthSinks {
-    download: Mutex<BandwidthSink>,
-    upload: Mutex<BandwidthSink>,
+    download: Atomic<u64>,
+    upload: Atomic<u64>,
```

Comment:
just curious, why not std Atomics? https://doc.rust-lang.org/core/sync/atomic/struct.AtomicU64.html

Reply:
> I used the atomic crate and an Atomic<u64> because the AtomicU64 type isn't available on all architectures. The atomic crate automatically falls back to using a Mutex on platforms that don't support AtomicU64.

sorry missed that

Reply:
`Atomic<u64>` is always a fallback on stable rust, something to be aware of
https://github.com/Amanieu/atomic-rs/blob/e52d9075341c4421f492e9c49a05b48068d8bb54/src/ops.rs#L21
> Only a subset of native atomic operations are supported on stable Rust (types which are the same size as AtomicUsize), but you can use the nightly Cargo feature on a nightly compiler to enable the full range of native atomic instructions. The nightly feature also enables const fn constructors which allow you to initialize static atomic variables.